docker-compose.yml

```diff
@@ -27,11 +29,11 @@ services:
         volumes:
             - type: bind
               source: ./frontend
-              target: /frontend
+              target: /app
             - /frontend/node_modules
             - /frontend/.parcel-cache
             - /frontend/.cache
-        working_dir: /frontend
+        working_dir: /app
     main:
```

Comment:
As I said before, these volumes are already in the develop branch, they are not my creation. I just fixed the path. If we do not want them, then Dan needs to approve this, as it is him who added them. I do not know the effects of removing the volumes.

Reply:
@danr you might want to comment your ideas on this matter.

Reply:
I don't know, I'm very confused about the docker volumes for the cache directories. If you're happy with this setup and don't run into caching problems, go for it!

Reply:
As I said in slack, what you do here is essentially removing the use of modules - they are still mounted under /frontend/node_modules etc. Whereas things are now run under /app (which will lead to the source directory being polluted by the cache folders and so on).

If you want to change to `/app`, I'd really want the volumes to be mounted there as well - but I suspect that will bring back your issue with the volume being corrupted over time (so I suspect you may want to run with the current setup in your specific environment).

Reply:
That might work, what was causing my compilation issues was having different volumen configurations in the dockerfile and the docker-compose. So you want to put everything inside the app folder instead? I am testing it now.

Reply:
I'm fairly convinced that what you ran into was a bug for Docker on macOS (which quite possibly can come back if you start using the volumes again). I don't much care about the actual path used inside the container, but I do not want temporary files and node libraries to spill outside of the container.

Reply:
In the original code, there was this line in the dockerfile:
VOLUME /app/node_modules

And then in the docker-compose:
volumes:
            - type: bind
              source: ./frontend
              target: /frontend
              - /frontend/node_modules

You are saying in one place that your volume in the container is under the /app folder and in the other that the same volume is under the /frontend folder. Do we agree on this?

Reply:
These would be independent (although possibly at some time in history intended to be the same, and that doesn't hurt but is not a strict requirement either).

The benefit of having the same path in both places is that the same volume is used (without going through to much trouble) and thus that the volume is "preloaded" on the first run of the instance. This means the `yarn install` and so on runs faster on the first launch.

Do note that for what happens in the container image, it's not unimportant whatever the `VOLUME` is used or not, having it will move what's under the given path to the new volume and thus it will not be included in the container image (it doesn't

That is not the same as the issue of changing some paths where things are mounted in the container but not other (e.g. changing the bind mount from `/frontend` to `/app` while still having the volumes at `/frontend/node_modules` and so on).

The `VOLUME` keyword in the Dockerfile will also register the volume for the container, so in theory, things should "just work" if one continue to use the same path. In practice, I've found that behaviour inconsistent, at least when having other mountpoints possibly interfering with it, so I prefer setting them up explicitly (which has been consistent in my experience).

Reply:
It seems like we are making some progress here.
Then, what do you want me to do?

Reply:
Anything that won't spill the contents that did go in the volume over to the bind mounted source directory is fine, this could be:

* reverting this change
* keeping the change but also changing the volumes so they are mounted as subdirectories in the same directory as the source  is mounted

Reply:
like this?
volumes:
            - type: bind
              source: ./frontend
              target: /app
            - ./frontend/node_modules:/app/node_modules
            - ./frontend/.parcel-cache:/app/.parcel-cache
            - ./frontend/.cache:/app/.cache

Reply:
IIRC that will be interpreted as additional bind mounts, which is what I don't want.

```
volumes:
  - type: bind
    source: ./frontend
    target: /app
  - /app/node_modules
  - /app/.parcel-cache
  - /app/.cache
```

should be fine.

Reply:
Great, it works. I have now pushed this version